Feature/connecting sql alchemy

bannybaks opened the pull request:
1. Alembic is connect

> Changes alembic.ini (path to migration scripts, template used to generate migration files), env.py (config.set_main_option; target_metadata);

2.  File *.python-version and requirements.py excluded from upload

> The requirements.py function is delegated to poetry

4. Container with Postgres database prepared

5. Prepared a script with commands for working with the cotainer

6. Added dependencies to poetry configuration, generate service file to run poetry

7. Create global project config

8. Described function create session

9.  Described base class for models

NiKuma0 commented on src/core/db/session.py:
```diff
@@ -0,0 +1,23 @@
+    id = Column(Integer, primary_key=True, nullable=False)
+
+
+Base = declarative_base()
```
Это не нужно для новой версии sqlalchemy...

NiKuma0 commented on src/core/db/session.py:
```diff
@@ -0,0 +1,23 @@
+
+Base = declarative_base()
+engine = create_async_engine(settings.database_url)
+AsyncSessionLocal = sessionmaker(engine, class_=AsyncSession)
```
функция `sessionmaker` возвращает не сессию, а фабрику (функция/класс который создает сессию). Поменяй название и помести в функцию `get_async_session`

NiKuma0 commented on src/settings.py:
```diff
@@ -0,0 +1,37 @@
+class Settings(BaseSettings):
+    """Global project settings."""
+
+    app_title: str = "RandomCoffeeBot"
+    description: str
```
Этого не надо.

NiKuma0 commented on src/settings.py:
```diff
@@ -0,0 +1,37 @@
+    app_title: str = "RandomCoffeeBot"
+    description: str
+
+    class Config:
```
Класс конфигов (или мета классы) всегда находится ниже всех методов и атрибутов.

NiKuma0 commented on src/settings.py:
```diff
@@ -0,0 +1,37 @@
+    class Config:
+        """Settings secrets"""
+
+        env_file = ".env"
```
В проде не будет env_file, этот конфиг только все поломает) Добавь переменную `TESTING` и в случае `True` добавь env_file.

NiKuma0 commented on src/settings.py:
```diff
@@ -0,0 +1,37 @@
+        """Settings secrets"""
+
+        env_file = ".env"
+        env_file_encoding = "utf-8"
```
Не думаю что это стоит явно указывать...

NiKuma0 commented on infra/Makefile:
Обычно Makefile в корень проекта ставят.

NiKuma0 commented on infra/Makefile:
```diff
@@ -0,0 +1,6 @@
+rm:
+	docker-compose -f docker-pg.yml stop && \
+	docker-compose -f docker-pg.yml rm && \
+	sudo rm -fr postgres_data
```
Что тут удаляется?) Насколько я вижу - postgres_data это docker volume, оно не так чистится...

NiKuma0 commented on src/core/db/models.py:
```diff
@@ -0,0 +1,7 @@
+
+class Base(DeclarativeBase):
+    """Base class for models"""
+
```
Сразу добавь `id`, `created_at` и `udpated_at`

bannybaks commented:
1. Corrected library versions; not relevant in prod moved to the dev section
2. Removed declarative base class; declarative_base() irrelevant for SQLAlchemy version 2.0; sessionmaker now returns a factory; the factory is transferred to the manager
3. Removed redundant code; permutatoin in the Settings class; using *.env by trigger; explicit *.env encoding declaration removed
4. Added fields id, created_at, updated_at
5. Move the Makefile to the root; command to delete free volumes
6. Removing everything related to pydantic

NiKuma0 commented on pyproject.toml:
```diff
@@ -6,16 +6,25 @@ authors = ["Павел Смирнов <baksbannysmirnov@yandex.ru>"]
+pre-commit = "^3.3.2"
+mako = "^1.2.4"
+typing-extensions = "^4.7.1"
+alembic = "^1.11.1"
```
Alembic должен быть в основной группе, не в dev

NiKuma0 commented on src/settings.py:
Убери весь этот файл

NiKuma0 reviewed (changes requested):
Расскажи как тестировал

bannybaks commented:
Тестирование:

1.  Локально поднял контейнер с базой
2. Подключился к базе через расширение для VSCode `Database Client`
3. Проверил что есть подключение
4. Создал тестовую таблицу
5. Проверил наличие полей `id`, `created_at`, `updated_at`

NiKuma0 commented:
> Тестирование:
> 
> 1. Локально поднял контейнер с базой
> 2. Подключился к базе через расширение для VSCode `Database Client`
> 3. Проверил что есть подключение
> 4. Создал тестовую таблицу
> 5. Проверил наличие полей `id`, `created_at`, `updated_at`

Это ты БД postgres проверил, а как же sqlalchemy? Нужно попробовать сделать запрос, создать модельку для тестов и проверить работу alembic-а
